### PR TITLE
Add openmm-hip

### PR DESCRIPTION
**Pull request by zklaus**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by ex-rzr** [comment]
I confirm that I want to be a maintainer.

**Comment by hmacdope** [comment]
I am willing to be a maintainer.

**Comment by zklaus** [comment]
LGTM. There may be room for future improvements, but I think it's reasonable to take that in the feedstock.

Of course I'll be very happy to add @peastman as a maintainer! But again, let's do that in the feedstock so it isn't a blocker for now.

Ready for review, @conda-forge/help-c-cpp.

Special ping for @ex-rzr, @hmacdope, @isuruf; feel free to comment here if I have not taken into account the discussion in https://github.com/openmm/openmm/issues/4199 appropriately or for any other reason.

**Comment by ex-rzr** [comment]
I just wanted to note that you can build with the defualt gcc (no need to pass `-DCMAKE_CXX_COMPILER=hipcc` and set `DEVICE_LIB_PATH`), this means that clangdev and lld (and, likely, compiler-rt) are also not required. However it seems that they are not as large as I expected (they are installed on the end user's system). Have you tried to build without them? This worked for me.

**Comment by zklaus** [comment]
Hm. I tried, but it didn't work for me (see [here](https://github.com/conda-forge/staged-recipes/runs/26886479466)). I am inclined to get this in as is and leave the switch to rtc to the feedstock. How do you feel about that?

**Comment by ex-rzr** [comment]
I suspect that only hipcc is needed because it contains some cmake stuff.

**Comment by ex-rzr** [comment]
Yes, seems so. Considering that it worked in my case I suggest one change to your commit: keep the `host: - hipcc`. I hope this is enough to make it work

**Comment by zklaus** [comment]
That seems to have worked, thanks @ex-rzr!

@conda-forge/help-c-cpp, this is ready for review.

**Comment by mikemhenry** [comment]
:wave: just a friendly bump; 

@conda-forge/help-c-cpp, this is ready for review.

File: recipes/openmm-hip/build.sh [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    ..
+
+make -j${CPU_COUNT}
+
+make install
+
+# Fix some overlinking warnings/errors
+for lib in ${PREFIX}/lib/plugins/libOpenMM*HIP*${SHLIB_EXT}; do
+    ln -s $lib ${PREFIX}/lib/$(basename $lib) || true
+done
```
Comment (isuruf) [inline recipes/openmm-hip/build.sh:23]: Are these lines really necessary?

File: recipes/openmm-hip/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    - hip-devel
+    - hipcc
+    - hipfft
+    - rocfft
+    - rocm-cmake
+    - rocm-device-libs
+    - openmm {{ version }}
+  run:
+    - {{ pin_compatible("hipfft") }}
+    - {{ pin_compatible("openmm") }}
```
Comment (isuruf) [inline recipes/openmm-hip/meta.yaml:37]:
```suggestion
    - {{ pin_compatible("openmm", max_pin="x.x.x") }}
```
?

**Comment by mikemhenry** [comment]
It looks like it is required 

<details>

```
2024-08-12T08:20:21.2796442Z WARNING: The install/build script(s) for openmm-hip deleted the following files (from dependencies) from the prefix:
2024-08-12T08:20:21.2798238Z ['etc/OpenCL/vendors/ocl-icd-system']
2024-08-12T08:20:21.2799648Z This will cause the post-link checks to mis-report. Please try not to delete and files (DSOs in particular) from the prefix
2024-08-12T08:20:21.4925369Z number of files: 25
2024-08-12T08:20:21.6033326Z Warning: rpath /home/conda/staged-recipes/build_artifacts/openmm-hip_1723450539572/_build_env/lib is outside prefix /home/conda/staged-recipes/build_artifacts/openmm-hip_1723450539572/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_ (removing it)
2024-08-12T08:20:21.6475317Z Warning: rpath /home/conda/staged-recipes/build_artifacts/openmm-hip_1723450539572/_build_env/lib is outside prefix /home/conda/staged-recipes/build_artifacts/openmm-hip_1723450539572/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_ (removing it)
2024-08-12T08:20:21.6613200Z Warning: rpath /home/conda/staged-recipes/build_artifacts/openmm-hip_1723450539572/_build_env/lib is outside prefix /home/conda/staged-recipes/build_artifacts/openmm-hip_1723450539572/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_ (removing it)
2024-08-12T08:20:21.6734135Z Warning: rpath /home/conda/staged-recipes/build_artifacts/openmm-hip_1723450539572/_build_env/lib is outside prefix /home/conda/staged-recipes/build_artifacts/openmm-hip_1723450539572/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_ (removing it)
2024-08-12T08:20:24.6612715Z    INFO: sysroot: '/home/conda/staged-recipes/build_artifacts/openmm-hip_1723450539572/_build_env/x86_64-conda-linux-gnu/sysroot/' files: '['usr/share/zoneinfo', 'usr/share/locale/zh_TW/LC_MESSAGES/libc.mo', 'usr/share/locale/zh_CN/LC_MESSAGES/libc.mo', 'usr/share/locale/vi/LC_MESSAGES/libc.mo']'
2024-08-12T08:20:24.6683661Z    INFO (openmm-hip,lib/plugins/libOpenMMDrudeHIP.so): Needed DSO lib/libOpenMM.so.8.1 found in conda-forge/linux-64::openmm==8.1.1=py312hce710af_1
2024-08-12T08:20:24.6697366Z   ERROR (openmm-hip,lib/plugins/libOpenMMDrudeHIP.so): $RPATH/libOpenMMHIP.so not found in packages, sysroot(s) nor the missing_dso_whitelist.
2024-08-12T08:20:24.6698088Z .. is this binary repackaging?
2024-08-12T08:20:24.6740589Z    INFO (openmm-hip,lib/plugins/libOpenMMDrudeHIP.so): Needed DSO lib/libOpenMMDrude.so found in conda-forge/linux-64::openmm==8.1.1=py312hce710af_1
2024-08-12T08:20:24.6779600Z    INFO (openmm-hip,lib/plugins/libOpenMMDrudeHIP.so): Needed DSO lib/libstdc++.so.6 found in conda-forge/linux-64::libstdcxx-ng==14.1.0=hc0a3c3a_0
2024-08-12T08:20:24.6801356Z    INFO (openmm-hip,lib/plugins/libOpenMMDrudeHIP.so): Needed DSO x86_64-conda-linux-gnu/sysroot/lib64/libm.so.6 found in CDT/compiler package conda-forge/noarch::sysroot_linux-64==2.17=h4a8ded7_16
2024-08-12T08:20:24.6840532Z    INFO (openmm-hip,lib/plugins/libOpenMMDrudeHIP.so): Needed DSO lib/libgcc_s.so.1 found in conda-forge/linux-64::libgcc-ng==14.1.0=h77fa898_0
2024-08-12T08:20:24.6863084Z    INFO (openmm-hip,lib/plugins/libOpenMMDrudeHIP.so): Needed DSO x86_64-conda-linux-gnu/sysroot/lib64/libc.so.6 found in CDT/compiler package conda-forge/noarch::sysroot_linux-64==2.17=h4a8ded7_16
2024-08-12T08:20:24.7252255Z    INFO (openmm-hip,lib/plugins/libOpenMMHIP.so): Needed DSO lib/libOpenMM.so.8.1 found in conda-forge/linux-64::openmm==8.1.1=py312hce710af_1
2024-08-12T08:20:24.7274093Z    INFO (openmm-hip,lib/plugins/libOpenMMHIP.so): Needed DSO x86_64-conda-linux-gnu/sysroot/lib64/libpthread.so.0 found in CDT/compiler package conda-forge/noarch::sysroot_linux-64==2.17=h4a8ded7_16
2024-08-12T08:20:24.7314174Z    INFO (openmm-hip,lib/plugins/libOpenMMHIP.so): Needed DSO lib/libhiprtc.so.6 found in conda-forge/linux-64::hip-runtime-amd==6.0.2=he02047a_3
2024-08-12T08:20:24.7352648Z    INFO (openmm-hip,lib/plugins/libOpenMMHIP.so): Needed DSO lib/libhipfft.so.0 found in conda-forge/linux-64::hipfft==1.0.13=hac33072_0
2024-08-12T08:20:24.7391068Z    INFO (openmm-hip,lib/plugins/libOpenMMHIP.so): Needed DSO lib/libamdhip64.so.6 found in conda-forge/linux-64::hip-runtime-amd==6.0.2=he02047a_3
2024-08-12T08:20:24.7429518Z    INFO (openmm-hip,lib/plugins/libOpenMMHIP.so): Needed DSO lib/libstdc++.so.6 found in conda-forge/linux-64::libstdcxx-ng==14.1.0=hc0a3c3a_0
2024-08-12T08:20:24.7451045Z    INFO (openmm-hip,lib/plugins/libOpenMMHIP.so): Needed DSO x86_64-conda-linux-gnu/sysroot/lib64/libm.so.6 found in CDT/compiler package conda-forge/noarch::sysroot_linux-64==2.17=h4a8ded7_16
2024-08-12T08:20:24.7489565Z    INFO (openmm-hip,lib/plugins/libOpenMMHIP.so): Needed DSO lib/libgcc_s.so.1 found in conda-forge/linux-64::libgcc-ng==14.1.0=h77fa898_0
2024-08-12T08:20:24.7509223Z    INFO (openmm-hip,lib/plugins/libOpenMMHIP.so): Needed DSO x86_64-conda-linux-gnu/sysroot/lib64/libc.so.6 found in CDT/compiler package conda-forge/noarch::sysroot_linux-64==2.17=h4a8ded7_16
2024-08-12T08:20:24.7528660Z    INFO (openmm-hip,lib/plugins/libOpenMMHIP.so): Needed DSO x86_64-conda-linux-gnu/sysroot/lib64/ld-linux-x86-64.so.2 found in CDT/compiler package conda-forge/noarch::sysroot_linux-64==2.17=h4a8ded7_16
2024-08-12T08:20:24.7614410Z    INFO (openmm-hip,lib/plugins/libOpenMMRPMDHIP.so): Needed DSO lib/libOpenMM.so.8.1 found in conda-forge/linux-64::openmm==8.1.1=py312hce710af_1
2024-08-12T08:20:24.7624971Z   ERROR (openmm-hip,lib/plugins/libOpenMMRPMDHIP.so): $RPATH/libOpenMMHIP.so not found in packages, sysroot(s) nor the missing_dso_whitelist.
2024-08-12T08:20:24.7625656Z .. is this binary repackaging?
2024-08-12T08:20:24.7665407Z    INFO (openmm-hip,lib/plugins/libOpenMMRPMDHIP.so): Needed DSO lib/libstdc++.so.6 found in conda-forge/linux-64::libstdcxx-ng==14.1.0=hc0a3c3a_0
2024-08-12T08:20:24.7684086Z    INFO (openmm-hip,lib/plugins/libOpenMMRPMDHIP.so): Needed DSO x86_64-conda-linux-gnu/sysroot/lib64/libm.so.6 found in CDT/compiler package conda-forge/noarch::sysroot_linux-64==2.17=h4a8ded7_16
2024-08-12T08:20:24.7722987Z    INFO (openmm-hip,lib/plugins/libOpenMMRPMDHIP.so): Needed DSO lib/libgcc_s.so.1 found in conda-forge/linux-64::libgcc-ng==14.1.0=h77fa898_0
2024-08-12T08:20:24.7742898Z    INFO (openmm-hip,lib/plugins/libOpenMMRPMDHIP.so): Needed DSO x86_64-conda-linux-gnu/sysroot/lib64/libc.so.6 found in CDT/compiler package conda-forge/noarch::sysroot_linux-64==2.17=h4a8ded7_16
2024-08-12T08:20:24.7842977Z    INFO (openmm-hip,lib/plugins/libOpenMMAmoebaHIP.so): Needed DSO lib/libOpenMM.so.8.1 found in conda-forge/linux-64::openmm==8.1.1=py312hce710af_1
2024-08-12T08:20:24.7858133Z   ERROR (openmm-hip,lib/plugins/libOpenMMAmoebaHIP.so): $RPATH/libOpenMMHIP.so not found in packages, sysroot(s) nor the missing_dso_whitelist.
2024-08-12T08:20:24.7858591Z .. is this binary repackaging?
2024-08-12T08:20:24.7895067Z    INFO (openmm-hip,lib/plugins/libOpenMMAmoebaHIP.so): Needed DSO lib/libOpenMMAmoeba.so found in conda-forge/linux-64::openmm==8.1.1=py312hce710af_1
2024-08-12T08:20:24.7933329Z    INFO (openmm-hip,lib/plugins/libOpenMMAmoebaHIP.so): Needed DSO lib/libstdc++.so.6 found in conda-forge/linux-64::libstdcxx-ng==14.1.0=hc0a3c3a_0
2024-08-12T08:20:24.7953551Z    INFO (openmm-hip,lib/plugins/libOpenMMAmoebaHIP.so): Needed DSO x86_64-conda-linux-gnu/sysroot/lib64/libm.so.6 found in CDT/compiler package conda-forge/noarch::sysroot_linux-64==2.17=h4a8ded7_16
2024-08-12T08:20:24.7991910Z    INFO (openmm-hip,lib/plugins/libOpenMMAmoebaHIP.so): Needed DSO lib/libgcc_s.so.1 found in conda-forge/linux-64::libgcc-ng==14.1.0=h77fa898_0
2024-08-12T08:20:24.8010821Z    INFO (openmm-hip,lib/plugins/libOpenMMAmoebaHIP.so): Needed DSO x86_64-conda-linux-gnu/sysroot/lib64/libc.so.6 found in CDT/compiler package conda-forge/noarch::sysroot_linux-64==2.17=h4a8ded7_16
```

</details>

File: recipes/openmm-hip/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    sha256: 6fc3547f39b8449773edff32170d842af1f5f60b8f2378c74ebc2261a50d9c14
+    folder: openmm-hip
+  - url: https://github.com/openmm/openmm/archive/refs/tags/{{ version }}.tar.gz
+    sha256: 347ad9f04dd88a673f7871127d9f23a75caf2c1a460a3f21f3328a24dc6547d0
+    folder: openmm
+
+
+build:
+  number: 0
+  skip: True  # [not linux]
```
Comment (isuruf) [inline recipes/openmm-hip/meta.yaml:19]:
```suggestion
  skip: True  # [not linux]
  rpaths:
    - lib
    - lib/plugins
```

**Comment by zklaus** [comment]
@conda-forge-admin, please lint

**Comment by zklaus** [comment]
The linting error is a bug in the linter (the location of the new style recipe.yaml is wrong), so everything seems fine now.

File: recipes/openmm-hip/meta.yaml (hunk 3 above)
Comment (zklaus) [reply recipes/openmm-hip/meta.yaml:19]: Much better solution. Thanks, @isuruf!

**Pull request merged**